src/blas/backends/cublas/cublas_batch.cpp

```diff
@@ -197,46 +584,282 @@ inline cl::sycl::event gemm_batch(Func func, cl::sycl::queue &queue, transpose *
+    throw unimplemented("blas", "trsm_batch", "for column_major layout");
 }
 
-cl::sycl::event axpy_batch(cl::sycl::queue &queue, int64_t *n, std::complex<double> *alpha,
-                           const std::complex<double> **x, int64_t *incx, std::complex<double> **y,
-                           int64_t *incy, int64_t group_count, int64_t *group_size,
+cl::sycl::event trsm_batch(cl::sycl::queue &queue, side left_right, uplo upper_lower,
+                           transpose trans, diag unit_diag, int64_t m, int64_t n,
+                           std::complex<double> alpha, const std::complex<double> *a, int64_t lda,
+                           int64_t stride_a, std::complex<double> *b, int64_t ldb, int64_t stride_b,
+                           int64_t batch_size,
                            const cl::sycl::vector_class<cl::sycl::event> &dependencies) {
-    throw unimplemented("blas", "axpy_batch", "for column_major layout");
+    throw unimplemented("blas", "trsm_batch", "for column_major layout");
+}
+
+template <typename Func, typename T>
+inline cl::sycl::event trsm_batch(Func func, cl::sycl::queue &queue, side *left_right,
+                                  uplo *upper_lower, transpose *trans, diag *unit_diag, int64_t *m,
+                                  int64_t *n, T *alpha, const T **a, int64_t *lda, T **b,
+                                  int64_t *ldb, int64_t group_count, int64_t *group_size,
+                                  const cl::sycl::vector_class<cl::sycl::event> &dependencies) {
+    using cuDataType = typename CudaEquivalentType<T>::Type;
+    for (int64_t i = 0; i < group_count; i++) {
+        overflow_check(m[i], n[i], lda[i], ldb[i], group_size[i]);
+    }
+    auto done = queue.submit([&](cl::sycl::handler &cgh) {
+        int64_t num_events = dependencies.size();
+        for (int64_t i = 0; i < num_events; i++) {
+            cgh.depends_on(dependencies[i]);
+        }
+        cgh.interop_task([=](cl::sycl::interop_handler ih) {
+            auto sc = CublasScopedContextHandler(queue);
+            auto handle = sc.get_handle(queue);
+            int64_t offset = 0;
+            cublasStatus_t err;
+            for (int64_t i = 0; i < group_count; i++) {
+                auto **a_ = reinterpret_cast<const cuDataType **>(a);
+                auto **b_ = reinterpret_cast<cuDataType **>(b);
+                CUBLAS_ERROR_FUNC(func, err, handle, get_cublas_side_mode(left_right[i]),
+                                  get_cublas_fill_mode(upper_lower[i]),
+                                  get_cublas_operation(trans[i]),
+                                  get_cublas_diag_type(unit_diag[i]), (int)m[i], (int)n[i],
+                                  (cuDataType *)&alpha[i], a_ + offset, (int)lda[i], b_ + offset,
+                                  (int)ldb[i], (int)group_size[i]);
+                offset += group_size[i];
+            }
+        });
+    });
+    return done;
+}
```

Comment:
Here, my goal was to take advantage of the fact that trsm_batch is supported in CUDA. (Other routines like axpy does not have batch implementations in CUDA) And since trsm_batch in CUDA requires only 1 group, I had to add a loop to handle multiple groups. (Similar to what has been done for gemm_batch)

Reply:
Yes i understand for trsm_batch but why not doing it for other functions?

Reply:
I don't think there is any technical reason for not doing it. It can be added by community. It is not in my priority list.